knimin/lib/data_access.py

```diff
@@ -3532,6 +3533,204 @@ def read_dna_plate(self, dna_plate_id):
+                          pm.plate_type USING(plate_type_id)
+                     WHERE shotgun_plate_id = %s"""
+            TRN.add(sql, [shotgun_plate_id])
+            res = dict(TRN.execute_fetchindex()[0])
+            n_rows = res['rows']
+            n_cols = res['cols']
+
+            if plate_normalization_sample.shape != (n_rows, n_cols):
+                obs = repr(plate_normalization_sample.shape)
+                exp = repr((n_rows, n_cols))
+                raise ValueError("plate_normalization_sample has an unexpected "
+                                 "shape %s; expected %s" % (obs, exp))
+
+            if plate_normalization_water.shape != (n_rows, n_cols):
+                obs = repr(plate_normalization_water.shape)
+                exp = repr((n_rows, n_cols))
+                raise ValueError("plate_normalization_water has an unexpected "
+                                 "shape %s; expected %s" % (obs, exp))
+
+            # create the instance of the normalized plate
+            sql = """INSERT INTO pm.shotgun_normalized_plate
+                        (shotgun_plate_id, created_on, email, echo_id)
+                     VALUES (%s, now(), %s, %s)
+                     RETURNING shotgun_normalized_plate_id"""
+
+            TRN.add(sql, [shotgun_plate_id, email, echo_id])
+            shotgun_normalized_plate_id = TRN.execute_fetchindex()[0][0]
+
+            # add all of the volume details for the normalized plate
+            sql = """INSERT INTO pm.shotgun_normalized_plate_well_values
+                        (shotgun_normalized_plate_id, row, col,
+                         sample_volume_nl, water_volume_nl)
+                     VALUES (%s, %s, %s, %s, %s)"""
+            sql_args = []
+            for row in np.arange(n_rows):
+                for col in np.arange(n_cols):
+                    sql_args.append((shotgun_normalized_plate_id,
+                                     row, col,
+                                     plate_normalization_sample[row, col],
+                                     plate_normalization_water[row, col]))
+            TRN.add(sql, sql_args, many=True)
+            TRN.execute()
+        return shotgun_normalized_plate_id
+
+    def read_normalized_shotgun_plate(self, shotgun_normalized_plate_id):
+        """Obtain the normalized shotgun plate details
+
+        Parameters
+        ----------
+        shotgun_normalized_plate_id : int
+            The ID of the normalized shotgun plate to retrieve.
+
+        Raises
+        ------
+        ValueError
+            If the normalized_shotgun_plate_id does not exist
+
+        Returns
+        -------
+        dict
+            A dict populated by the columns details in
+            pm.shotgun_normalized_plate, as well as two numpy matrices
+            corresponding to the normalized water and sample volumes
+        """
+        with TRN:
+            # get plate details
+            sql = """SELECT *
```

Comment:
@wasade can you be specific on the request here? Something similar as what is done [here](https://github.com/biocore/labadmin/blob/plate-mapper-kiwi/knimin/lib/data_access.py#L3977).

I know the ID is redundant but we've included it in all the other alike calls, it simplifies a few calls done later.
Also, instead of returning the IDs of the different attributes of the plates, can you return the names? These would be the name of: echo, shotgun_library_prep_kit, shotgun_adapter_aliquot and qpcr